sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3381,9 +3381,13 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    * }}}
    */
   override def visitShowPartitions(ctx: ShowPartitionsContext): LogicalPlan = withOrigin(ctx) {
-    val table = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val partitionKeys = Option(ctx.partitionSpec).map(visitNonOptionalPartitionSpec)
-    ShowPartitionsStatement(table, partitionKeys)
+    val multiPart = Option(ctx.multipartIdentifier).map(visitMultipartIdentifier)
+    val partitionKeys = Option(ctx.partitionSpec).map { specCtx =>
+      UnresolvedPartitionSpec(visitNonOptionalPartitionSpec(specCtx), None)
+    }
+    ShowPartitions(
+      UnresolvedTableOrView(multiPart.getOrElse(Seq.empty[String])),
```

Comment:
If it doesn't support view, why not use `UnresolvedTable`?

Reply:
To output proper error as for V1.

Reply:
see https://github.com/apache/spark/pull/30398/files#diff-583171e935b2dc349378063a5841c5b98b30a2d57ac3743a9eccfe7bffcb8f2aR1011-R1012

Reply:
V1: https://github.com/apache/spark/blob/23e9920b3910e4f05269853429c7f18888cdc7b5/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L992-L994

Reply:
then we should improve the v2 error message like https://github.com/apache/spark/pull/30475

Reply:
One of purpose of this PR is to create v2 implementation compatible to v1. I would prefer to avoid changing v1 impl at least in this PR since v1 is out of the scope of this PR.

Reply:
- I replaced by `UnresolvedTableOrView` by `UnresolvedTable`, and
- removed some checks from `CheckAnalysis.checkShowPartitions()`